### PR TITLE
Remove the attemptId notion in the connectionManagerWorkflow

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
This is removing the attemptId from the create attempt activity to replace it with the attemptNumber. This will be modified in the workflow in a later commit.

### CODE REVIEW

**Comment by benmoriceau** [comment]
> Benoit, can you explain more why replacing the attempt id with the attempt number is better? Probably a dumb question, what's the difference between the attempt id and the attempt number?

It is to help understanding what represent the attemptId/Attempt number that we have in the temporal workflow. The doc in `airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/JobPersistence.java` was missleading because the implementation is returning the attemptNumber instead of the attemptId. The goal is to remove the confusion because here attemptId we use in the temporal workflow is actually an attempt number.

Here is the persistence implementation (just the return that select the number):
```
return ctx.fetch(
          "INSERT INTO attempts(job_id, attempt_number, log_path, status, created_at, updated_at) VALUES(?, ?, ?, CAST(? AS ATTEMPT_STATUS), ?, ?) RETURNING attempt_number",
          jobId,
          job.getAttemptsCount(),
          logPath.toString(),
          Sqls.toSqlName(AttemptStatus.RUNNING),
          now,
          now)
          .stream()
          .findFirst()
          .map(r -> r.get("attempt_number", Integer.class))
          .orElseThrow(() -> new RuntimeException("This should not happen"));
    });
```

**Comment by benmoriceau** [comment]
Move it back to draft, I planned to do it in multiple PR but it doesn't make sense to have that in multiple version tag.

**Comment by davinchia** [comment]
Thanks for the explanation. Please tag me again once this is ready for review.

**Comment by benmoriceau** [comment]
@davinchia it is ready to review

The version stuff and the forced renaming is pretty ugly (unfortunately temporal doesn't allow override). @pmossman do you see a better approach to that?

**Comment by benmoriceau** [comment]
I tested the version increase locally by launching a long running job from master branch and then checkout the branch of this review and restart the docker compose.

**Review by pmossman — APPROVED** [review]
Discussed a bit with Benoit via Slack. I think there is perhaps an approach we could take that would avoid needing to create all of the redundant activity methods, but it might be risky.

Basically, instead of adding a new activity method that takes in a new activity input, we might be able to add a nullable `attemptNumber` field to the existing activity input classes. 

I _think_ the workflow replayer can handle new nullable fields on activity inputs/outputs, so we could theoretically avoid needing to create the new activity methods at all. Then later on when all workflows using the old field are closed, we could remove it without any workflow versioning.

This is definitely risky and might not be worth trying since we really have to avoid stuck workflows. If we went this route, we should definitely try to verify that it works with the https://cadenceworkflow.io/docs/java-client/workflow-replay-shadowing/#workflow-replayer tool. I'll leave it up to you to decide what makes the most sense for this particular PR.

Here's a slack discussion that Brodie (former teammate at LiveRamp) had with Maxim about this topic: 
image

File: airbyte-workers/build.gradle
Comment (davinchia) [inline airbyte-workers/build.gradle:36]: what happened to this?

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/state/WorkflowInternalState.java
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/state/WorkflowInternalState.java:22]: nit: can we remove this?

**Review by davinchia — COMMENTED** [review]
Thanks!

Thinking about this more, I really like the idea @pmossman brought up. Given that versioning is tricky in Temporal, I feel strongly we should explore this option before we go ahead and create yet another Temporal state object. It sounds like we have time to do so (this doesn't seem urgent) but definitely lmk if I'm wrong. Holding off approval until we come to a good decision.

Small feedback here - in the future can we please leave the usual PR template and make sure to fill in why, how and the various files to look at? I know it can be annoying, however it really does help make the PR much easier to digest. Thanks!

File: airbyte-workers/build.gradle
Comment (benmoriceau) [reply airbyte-workers/build.gradle:36]: I think that it is not needed. The `temporak-testing-junit5` version doesn't go above 1.5 we should only keep the `temporak-testing` to be align with the temporal sdk

**Comment by benmoriceau** [comment]
> Thanks!
> 
> Thinking about this more, I really like the idea @pmossman brought up. Given that versioning is tricky in Temporal, I feel strongly we should explore this option before we go ahead and create yet another Temporal state object. It sounds like we have time to do so (this doesn't seem urgent) but definitely lmk if I'm wrong. Holding off approval until we come to a good decision.
> 
> Small feedback here - in the future can we please leave the usual PR template and make sure to fill in why, how and the various files to look at? I know it can be annoying, however it really does help make the PR much easier to digest. Thanks!

Will keep keeping the template in mind.

About Parker suggestion, I have added `WorkflowReplayingTest` that is doing that. It currently has a single scenario, we should add more and have it being autogenerate.

File: airbyte-workers/build.gradle
Comment (davinchia) [reply airbyte-workers/build.gradle:36]: sounds good.

**Pull request merged**
